### ISSUE TITLE
Revise BAIP-01: Purpose and Guidelines

### ISSUE THREAD

**Issue by abitmore** [bitshares/baips#11]
Just my own opinion, the voting process (BAIP-threshold) should not be in the purpose and guidelines, but in a new BAIP, because it's a bit controversial. It's better to have a simple and relatively stable document as "purpose and guidelines".

I was listed as one of the editors, although I'd like to help when I have time, I don't always have time to respond quickly enough. It's better to have someone more suitable here, also good if we have guidelines about how to choose new editors.

Redefine the process. E.G. we're heavily relying on Github to collaborate, but not sending emails back and forth.

**Comment by abitmore** [comment]
I think there should a new status "ready for voting" after "draft", or a new status "pre-draft" before "draft".

**Comment by abitmore** [comment]
The template mentioned in BSIP01 doesn't exist in this repository.

**Comment by zhouxiaobao-2010** [comment]
I also think there are some loose spots. But because the proposal of feed price reform is urgent, I approved it first so as to enter the voting process as soon as possible. I think we have time to improve the mechanism of this BAIP later. I hope you can discuss more.

**Comment by zhouxiaobao-2010** [comment]
I agree with Abit. Maybe a new BAIP vote is needed to determine the threshold. Before that, we can still follow the previous practice—— more than 400k is deemed to be passed.

**Comment by bangzi1001** [comment]
>Just my own opinion, the voting process (BAIP-threshold) should not be in the purpose and guidelines, but in a new BAIP, because it's a bit controversial. It's better to have a simple and relatively stable document as "purpose and guidelines".

BSIP 1 does not define voting process clearly and causing lot of unnecessary drama. I would say BAIP 1 have a clear and understandable voting process. It may not be the best voting rule at this moment, but committee members can redefine the voting process if they are able to find a better voting process.

>I was listed as one of the editors, although I'd like to help when I have time, I don't always have time to respond quickly enough. It's better to have someone more suitable here, also good if we have guidelines about how to choose new editors.

As the asset owner of BitAssets, all Bitshares active committee members should have privilege to this repo or naturally become one of the editors. Bitcrab, Xeroc and you are listed as editors because you guys voted in as committee members since very long time ago, others are easily voted in and out. There is no point to edit BAIP 1 too often. There is also no point to choose new editor purposely, the appointed editors may think they have higher authority than committee members, this will create drama like BSIP76.

**Comment by grctest** [comment]
> Bitshares active committee members should have privilege to this repo or naturally become one of the editors.

If this is made to be the case, if one has a vested interest in a BAIP the editor role aught be delegated to non immediately involved parties, no?

We've seen with https://github.com/bitshares/baips/pull/9 bypass proposed BAIP processes, peer review of cnvote proposed BAIP/BSIP seems to not be tolerated.

**Comment by abitmore** [comment]
The description about "accepted" status in BAIP-01 is confusing:
> People wishing to submit BAIPs first should propose their idea as github issue first. After discussion you will be assigned a number for the BAIP and can send a pull request for your draft. Once consensus among discussion participants is reached, the status can be switched to *accepted*. From this time on, major changes of the document will not be permitted.
> ...
> Once a BAIP has been published, the reference implementation must be completed. When the reference implementation is complete and accepted by the BTS holders via approval voting, the status will be changed to **"Accepted"**. A BAIP can also be "Rejected" by BTS holders.

Also posted the same question to https://github.com/bitshares/bsips/issues/250#issuecomment-551888755.

**Comment by clockworkgr** [comment]
> I think there should a new status "ready for voting" after "draft", or a new status "pre-draft" before "draft".

I recommended on telegram that we need a "Ready for voting" status on BAIPs that are considered ready for voting.
After voting concludes, BAIPs can be either Accepted or Rejected. 

Exact status titles can be different but totally agree with you

**Comment by bangzi1001** [comment]
Beside "Accepted" status, I add a new status "Accepted for X(eg.BitCNY) Only" in BAIPs-Template.md because because some BAIP may apply to a specific BitAsset eg. BitUSD but not all BitAssets.

**Comment by sschiessl-bcp** [comment]
> > I think there should a new status "ready for voting" after "draft", or a new status "pre-draft" before "draft".
> 
> I recommended on telegram that we need a "Ready for voting" status on BAIPs that are considered ready for voting.
> After voting concludes, BAIPs can be either Accepted or Rejected.
> 
> Exact status titles can be different but totally agree with you

Accepted and Rejected is fine, since a "Ready for Voting" status carries no meaning for informal BAIPs. I did not see that it also talks about status "Accepted" after voting. 

I would suggest to refine the wording:

Status | Meaning | Action for Authors | Action for Editors
--- | --- | --- | ---
Exists only as issue | Idea is maturing and discussion ongoing | Flesh out the text and address or incorporate feedback, describe all concerns in the document to allow a holistic opinion making for future readers / voters | Guide authors and give feedback, once considered mature, open Pull Request with Status "Draft" (may also be opened by Authors)
Draft | Discussion is still ongoing or review in progress | Implement change requests from reviewers | Review, and once considered completed change Status to "Accepted" and merge (may also be merged with Status "Draft" to allow preview, needs a subsequent Pull Request to refinement and Status change)
Accepted | This BAIP is considered ready for voting | Create Poll Worker Proposals (by itself or through escrow) | Pending the voting outcome, change Status to "Approved" (voting was positive) or "Rejected" (voting was negative)
Approved | This BAIP is approved by BTS holders and is to be realized | Nothing | Nothing
Rejected | This BAIP was rejected by BTS holders. | Nothing | Nothing
Installed | This BSIP required a protocol upgrade and is live on mainnet. | Nothing | Nothing

**Comment by sschiessl-bcp** [comment]
Any thoughts?

**Comment by shulthz** [comment]
I think need to add some rules make the process clearly:

> Must make the BSIP process clearly :
> 
> Write a BSIP → Discuss and comment → Request a BSIP number → **Deliberated and assigned a BSIP number by the committees( or grantee of committees) , they can vote for this**→ BSIP status: Accepted → Voted by holders of BTS → BSIP status: Approved → Development → Pull request → Merge..
> 
> **Settlement of Disputes**
> 
> If somebody or some team questions the action of BSIP by committees( or grantee of committees), they can ask a committees vote which must be submitted by three committees together at least, or they didn't believe the committees, they can ask a witness vote which must be submitted by five witnesses together at least.
